### PR TITLE
fix: fix trailColor in line-type progress and add trailColor in docs

### PR DESCRIPTION
**Pull request by AshoneA**
### 🤔 This is a ...

- [] New feature
- [x] Bug fix
- [x] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/21550

### 💡 Background and solution
fix `trailColor` props in `type=line` progress and add trailColor in docs

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix Progress `trailColor` not working when `type=line`.        |
| 🇨🇳 Chinese |     修复 Progress `trailColor`属性在 `type=line` 时无效果的问题。    |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/progress/demo/line.md](https://github.com/AshoneA/ant-design/blob/fix/progress-trail-color/components/progress/demo/line.md)
[View rendered components/progress/index.en-US.md](https://github.com/AshoneA/ant-design/blob/fix/progress-trail-color/components/progress/index.en-US.md)
[View rendered components/progress/index.zh-CN.md](https://github.com/AshoneA/ant-design/blob/fix/progress-trail-color/components/progress/index.zh-CN.md)

### CODE REVIEW

File: components/progress/demo/line.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ ReactDOM.render(
     <Progress percent={70} status="exception" />
     <Progress percent={100} />
     <Progress percent={50} showInfo={false} />
+    <Progress percent={30} trailColor="#1890ff30" />
```
Comment (afc163) [inline components/progress/demo/line.md]: 演示中可以不展示。

File: components/progress/Line.tsx [hunk 2]
```diff
@@ -69,15 +69,22 @@ const Line: React.SFC<LineProps> = props => {
     strokeColor,
     strokeLinecap,
     children,
+    trailColor,
   } = props;
   let backgroundProps;
+  let trailStyle;
```
Comment (afc163) [inline components/progress/Line.tsx]: 定义应靠近使用的地方。

File: components/progress/demo/line.md (hunk 1 above)
Comment (AshoneA) [reply components/progress/demo/line.md]: 已去掉

File: components/progress/Line.tsx (hunk 2 above)
Comment (AshoneA) [reply components/progress/Line.tsx]: 已定义于到最靠近使用的地方

**Comment by afc163** [comment]
加个用例把覆盖率提起来。

**Comment by AshoneA** [comment]
@afc163 `trailColor`的测试用例已加上, 写测试用例时候发现`line.tsx`还有一处测试用例没覆盖到, 然后看了下感觉里面的有段代码有点问题. pr在这里 https://github.com/ant-design/ant-design/pull/21564 , 加上这个pr中的`line.tsx`覆盖率已经达到100%了

**Comment by afc163** [comment]
@AshoneA Well done!

**Pull request merged**